src/libraries/reduxUtils/convert.ts

```diff
@@ -1,17 +1,44 @@
-import moment from "moment";
+import { MedicalDTO } from "../../generated";
+import { renderDate } from "../formatUtils/dataFormatting";
 import { SummaryFieldType } from "./SummaryFieldType";
 
 export const convertToSummaryData = (
   data: Array<any>,
   field: SummaryFieldType
 ) => {
   const res = data.map(({ ...rest }) => ({
+    ...rest,
     type: field.type,
-    date: moment(+rest[field.dateField]).isValid()
-      ? moment(+rest[field.dateField]).format("DD/MM/YYYY")
-      : "",
-    note: field.noteField ? rest[field.noteField] : "",
-    result: "",
+    date: rest[field.dateField],
   }));
   return res;
 };
+
+export const renderSummary = (
+  data: Array<any>,
+  dateFields: string[],
+  labels: any,
+  medicals: MedicalDTO[] = []
+) => {
+  const itemRender = (item: any) => {
+    const obj: any = {};
+    Object.keys(labels).map((field: string) => {
+      if (typeof item[field] === "object") {
+        obj[field] = item[field]?.description ?? "";
+      } else if (dateFields.includes(field) && item[field]) {
+        obj[field] = renderDate(item[field]);
+      } else if (field === "medicalId" && item[field]) {
+        obj[field] =
+          medicals.find((medoc) => medoc.code === item[field])?.description ??
+          item[field];
+      } else if (item[field]) {
+        obj[field] = item[field];
+      }
+      return obj[field];
+    });
+    return obj;
+  };
+  return data.map((item: any) => {
+    return itemRender(item);
+  });
+};
```

Comment:
I'm not sure about this. This method is both not generic and not specific. Are you trying to make a generic function (using any) but if you go ahead with the workflow I see some specific property (.description). I think that we could add some type in order to make this method works as expected with typescript.

Reply:
You are right. I use a combination of type there. It's OK.